engine/eConstr.ml

```diff
@@ -651,6 +654,10 @@ let substnl subst n c = of_constr (Vars.substnl (List.map to_constr subst) n (to
 let substl subst c = of_constr (Vars.substl (List.map to_constr subst) (to_constr c))
 let subst1 c r = of_constr (Vars.subst1 (to_constr c) (to_constr r))
 
+let substnl_decl subst n d = of_rel_decl (Vars.substnl_decl (List.map to_constr subst) n (to_rel_decl d))
```

Comment:
IMO we are using very dangerous terminology here: a "defined evar" is not an evar, but whatever term the substitution pointed to; "evar insensitivity" has nothing to do with "sensitivity" or "evars" but with a the critical bug of failing to propagate a substitution.

The key point is that there is not a single case where "exposing defined evars" that is "failing to apply a substitution" can be acceptable, however we seem to like to legitimize that broken behavior. It is like pretending `(λ x . x) 2 → x` is a valid reduction sometimes.